### PR TITLE
Implementation of the Orientation Filter

### LINKED ISSUE #21: Implement Orientation KF

### PR DESCRIPTION
**Pull request by jbinz**
This patch implements the orientation Filter. The Filter is working, however there are still some irregularities which I think are due to the magnetometer not being calibrated.

The new filter improves the orientation Kalman Filter in every aspect:
State estimation task processor usage with KF: 6.5%		free memory: 100 words
State estimation task processor usage with new filter: 4%	free memory: 600 words
Additionally the new filter takes into account that the acceleration might be very large, rendering the accel measurement useless.
It also seems to be much more stable.
Needs to be verified on a drone and on a rocket.
Closes #21

### CODE REVIEW

File: boards/cats_rev1Pro/Core/Inc/control/orientation_filter.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+} orientation_filter_t;
+
+/* Orientation Kalman Filter */
+
+#define INIT_COV   0.1f
+#define NOISE_VEL  0.01f
+#define NOISE_POS  0.03f
+#define NOISE_BIAS 0.01f
+
+typedef struct {
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/Core/Inc/control/orientation_filter.h:59]: I would put ifdefs around both of these struct definitions

File: boards/cats_rev1Pro/CMakeLists.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #THIS FILE IS AUTO GENERATED FROM THE TEMPLATE! DO NOT CHANGE!
 set(CMAKE_SYSTEM_NAME Generic)
 set(CMAKE_SYSTEM_VERSION 1)
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.19)
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/CMakeLists.txt:4]: revert this to 3.16, I had some problems installing newer versions on my PC :)

File: boards/cats_rev1Pro/Core/Src/control/quaternion.c [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  output[2] = input[2];
+  output[3] = input[3];
+  output[4] = 0;
+  output[5] = -input[1];
+  output[6] = -input[2];
+  output[7] = input[1];
+  output[8] = 0;
+}
+
+/* ALL INPUTS AND OUTPUTS NEED TO BE 4,1 MATRIXES */
```
Comment (stojadin2701) [inline boards/cats_rev1Pro/Core/Src/control/quaternion.c:20]: Matrices

**Review by stojadin2701 — COMMENTED** [review]
Minor comments

**Review by stojadin2701 — APPROVED** [review]

**Review by l-jost — CHANGES_REQUESTED** [review]

**Review by l-jost — APPROVED** [review]

**Pull request merged**